File: lab/python/medias/media_array.py
def media1(*valores):
    if not valores:
        return None
    return sum(valores) / len(valores)


def media2(valores):
    if not valores or not isinstance(valores, list):
        return None
    return sum(valores) / len(valores)


def media3(valores):
    if not valores or not isinstance(valores, list):
        return None

    soma = 0
    for valor in valores:
        soma += valor

    return soma / len(valores)


def media4(valores):
    if not valores or not isinstance(valores, list):
        return None

    soma = 0
    for valor in valores:
        soma += valor

    total = 0
    for _ in valores:
        total += 1

    return soma / total


lista = [10, 20, 30]
print(media1(*lista))
print(media2(lista))
print(media3(lista))
print(media4(lista))

File: lab/javascript/medias/media_array.js
function media1(...valores) {
  if (valores.length === 0) return null;
  const soma = valores.reduce((a, b) => a + b, 0);
  return soma / valores.length;
}

function media2(valores) {
  if (!Array.isArray(valores) || valores.length === 0) return null;
  const soma = valores.reduce((a, b) => a + b, 0);
  return soma / valores.length;
}

function media3(valores) {
  if (!Array.isArray(valores) || valores.length === 0) return null;

  let soma = 0;
  for (let valor of valores) {
    soma += valor;
  }
  return soma / valores.length;
}

function media4(valores) {
  if (!Array.isArray(valores) || valores.length === 0) return null;

  let soma = 0;
  for (let valor of valores) {
    soma += valor;
  }

  let total = 0;
  for (let _ of valores) {
    total += 1;
  }

  return soma / total;
}

// -------------------- Teste --------------------
const lista = [10, 20, 30];
console.log(media1(...lista)); // 20
console.log(media2(lista));    // 20
console.log(media3(lista));    // 20
console.log(media4(lista));    // 20
